### PR TITLE
Fixes #1025 DialogBoxAreYouSure not working anymore

### LINKED ISSUE #1025: Deleting save results in UnassignedReferenceException
To reproduce:
- open load dialog
- click on a save
- hit the red x

Callstack:
```
UnassignedReferenceException: The variable dialog of DialogBoxLoadGame has not been assigned.
You probably need to assign the dialog variable of the DialogBoxLoadGame script in the inspector.
DialogBoxLoadGame.DeleteWasClicked () (at Assets/Scripts/UI/DialogBox/FileSaveLoad/DialogBoxLoadGame.cs:114)
UnityEngine.Events.InvokableCall.Invoke (System.Object[] args) (at C:/buildslave/unity/build/Runtime/Export/UnityEvent.cs:153)
UnityEngine.Events.InvokableCallList.Invoke (System.Object[] parameters) (at C:/buildslave/unity/build/Runtime/Export/UnityEvent.cs:630)
UnityEngine.Events.UnityEventBase.Invoke (System.Object[] parameters) (at C:/buildslave/unity/build/Runtime/Export/UnityEvent.cs:765)
UnityEngine.Events.UnityEvent.Invoke () (at C:/buildslave/unity/build/Runtime/Export/UnityEvent_0.cs:53)
UnityEngine.UI.Button.Press () (at C:/buildslave/unity/build/Extensions/guisystem/UnityEngine.UI/UI/Core/Button.cs:35)
UnityEngine.UI.Button.OnPointerClick (UnityEngine.EventSystems.PointerEventData eventData) (at C:/buildslave/unity/build/Extensions/guisystem/UnityEngine.UI/UI/Core/Button.cs:44)
UnityEngine.EventSystems.ExecuteEvents.Execute (IPointerClickHandler handler, UnityEngine.EventSystems.BaseEventData eventData) (at C:/buildslave/unity/build/Extensions/guisystem/UnityEngine.UI/EventSystem/ExecuteEvents.cs:52)
UnityEngine.EventSystems.ExecuteEvents.Execute[IPointerClickHandler] (UnityEngine.GameObject target, UnityEngine.EventSystems.BaseEventData eventData, UnityEngine.EventSystems.EventFunction`1 functor) (at C:/buildslave/unity/build/Extensions/guisystem/UnityEngine.UI/EventSystem/ExecuteEvents.cs:269)
UnityEngine.EventSystems.EventSystem:Update()
```

I didn't fully investigate the issue, this may be a very easy fix.

### PR DESCRIPTION
**Pull request by sboigelot**
Fixes #1025

### CODE REVIEW

File: Assets/Scripts/UI/DialogBox/YesNo/DialogBoxAreYouSure.cs [hunk 1]
```diff
@@ -6,18 +6,46 @@
...
+
+    public override void CloseDialog()
     {
+        InvodeClosed();
+        base.CloseDialog();
     }
 
-    // Update is called once per frame
-    private void Update()
+    private void InvodeClosed()
```
Comment (bjubes) [inline Assets/Scripts/UI/DialogBox/YesNo/DialogBoxAreYouSure.cs]: typo? `InvokeClosed`?

Comment (sboigelot) [inline Assets/Scripts/UI/DialogBox/YesNo/DialogBoxAreYouSure.cs]: fixed :)

**Comment by vogonistic** [comment]
You should name the title of the PR so that people don't have to click though to an issue to figure out what is going on. It's great to have fixes in the body, but it's a bad title.

**Comment by sboigelot** [comment]
@vogonistic indeed, I didn't think about it, I'll update my 2 pending fixes

**Comment by koosemose** [comment]
Looks good, no unexpected bugs, Stylecop clean. Merged.

**Pull request merged**